File: cases/taint/kotlin/return-relay-two-hop-positive/ReturnRelayTwoHopPositive.kt
package dataflowbench

object ReturnRelayTwoHopPositive {
    fun dfb_source(): Int { // DFB-SOURCE: return-two-hop-input
        return 1
    }

    fun firstRelay(value: Int): Int { // DFB-WITNESS: return-two-hop-first
        return value
    }

    fun secondRelay(value: Int): Int { // DFB-WITNESS: return-two-hop-second
        return firstRelay(value)
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: return-two-hop-sink

    fun run() {
        val result = secondRelay(dfb_source())
        dfb_sink(result)
    }
}

File: cases/taint/go/return-relay-two-hop-positive/return_relay_two_hop_positive.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: return-two-hop-input
	return 1
}

func firstRelay(value int) int { // DFB-WITNESS: return-two-hop-first
	return value
}

func secondRelay(value int) int { // DFB-WITNESS: return-two-hop-second
	return firstRelay(value)
}

func dfb_sink(value int) {} // DFB-SINK: return-two-hop-sink

func run() {
	result := secondRelay(dfb_source())
	dfb_sink(result)
}
